openapi2kong/openapi2kong.go

```diff
@@ -53,263 +56,209 @@ func (opts *O2kOptions) setDefaults() {
 // returns nil if it wasn't found, an error if it wasn't an object or couldn't be
 // dereferenced. The returned object will be json encoded again.
-func getXKongObject(props openapi3.ExtensionProps, key string, components *map[string]interface{}) ([]byte, error) {
-	if props.Extensions != nil && props.Extensions[key] != nil {
-		var jsonBlob interface{}
-		_ = json.Unmarshal(props.Extensions[key].(json.RawMessage), &jsonBlob)
-		jsonObject, err := jsonbasics.ToObject(jsonBlob)
-		if err != nil {
-			return nil, fmt.Errorf("expected '%s' to be a JSON object", key)
-		}
+func getXKongObject(
+	extensions *orderedmap.Map[string, *yaml.Node],
+	key string, components *map[string]interface{},
+) ([]byte, error) {
+	if extensions == nil {
+		return nil, nil
+	}
 
-		object, err := dereferenceJSONObject(jsonObject, components)
-		if err != nil {
-			return nil, err
-		}
-		return json.Marshal(object)
+	xKongObject, ok := extensions.Get(key)
+	if !ok || xKongObject == nil {
+		return nil, nil
 	}
-	return nil, nil
+
+	xKongObjectBytes, err := convertYamlNodeToBytes(xKongObject)
+	if err != nil {
+		return nil, fmt.Errorf("expected '%s' to be a YAML object", key)
+	}
+
+	var jsonBlob interface{}
+	_ = yaml.Unmarshal(xKongObjectBytes, &jsonBlob)
+	jsonObject, err := jsonbasics.ToObject(jsonBlob)
+	if err != nil {
+		return nil, fmt.Errorf("expected '%s' to be a JSON/YAML object", key)
+	}
+
+	object, err := dereferenceJSONObject(jsonObject, components)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(object)
 }
 
 // getXKongComponents will return a map of the '/components/x-kong/' object. If
 // the extension is not there it will return an empty map. If the entry is not a
-// Json object, it will return an error.
-func getXKongComponents(doc *openapi3.T) (*map[string]interface{}, error) {
+// yaml object, it will return an error.
+func getXKongComponents(doc v3.Document) (*map[string]interface{}, error) {
 	var components map[string]interface{}
-	switch prop := doc.Components.ExtensionProps.Extensions["x-kong"].(type) {
-	case nil:
-		// not available, create empty map to do safe lookups down the line
-		components = make(map[string]interface{})
 
-	default:
-		// we got some json blob
-		var xKong interface{}
-		_ = json.Unmarshal(prop.(json.RawMessage), &xKong)
+	if doc.Components == nil || doc.Components.Extensions == nil {
+		return &map[string]interface{}{}, nil
+	}
 
-		switch val := xKong.(type) {
-		case map[string]interface{}:
-			components = val
+	xKongComponents, ok := doc.Components.Extensions.Get("x-kong")
 
-		default:
-			return nil, fmt.Errorf("expected '/components/x-kong' to be a JSON object")
-		}
+	if !ok || xKongComponents == nil {
+		return &components, nil
+	}
+
+	xKongComponentsBytes, err := convertYamlNodeToBytes(xKongComponents)
+	if err != nil {
+		return nil, fmt.Errorf("expected '/components/x-kong' to be a YAML object")
+	}
+
+	var xKong interface{}
+	_ = yaml.Unmarshal(xKongComponentsBytes, &xKong)
```

Comment:
same question about the error handling

Reply:
Same reason as above error handling comment. I have changed this one's handling to how we handle things above as well.